crates/aiken-lang/src/ast.rs

```diff
@@ -909,26 +909,26 @@ pub type MultiPattern<PatternConstructor, Type> = Vec<Pattern<PatternConstructor
 pub type UntypedMultiPattern = MultiPattern<(), ()>;
 pub type TypedMultiPattern = MultiPattern<PatternConstructor, Arc<Type>>;
 
-pub type TypedClause = Clause<TypedExpr, PatternConstructor, Arc<Type>>;
-pub type UntypedClause = Clause<UntypedExpr, (), ()>;
+#[derive(Debug, Clone, PartialEq)]
+pub struct UntypedClause {
+    pub location: Span,
+    pub patterns: Vec1<Pattern<(), ()>>,
+    pub guard: Option<ClauseGuard<()>>,
+    pub then: UntypedExpr,
+}
 
 #[derive(Debug, Clone, PartialEq)]
-pub struct Clause<Expr, PatternConstructor, Type> {
+pub struct TypedClause {
     pub location: Span,
-    pub pattern: MultiPattern<PatternConstructor, Type>,
-    pub alternative_patterns: Vec<MultiPattern<PatternConstructor, Type>>,
-    pub guard: Option<ClauseGuard<Type>>,
-    pub then: Expr,
+    pub pattern: Pattern<PatternConstructor, Arc<Type>>,
```

Comment:
Note: I decided to also make this desugaring more apparent in the typed AST. So here indeed, we can see that `TypedClause` can only have a single pattern. We desugar during type-check. It makes things slightly less awkward down the line -- in particular in code gen which was already assuming anyway that clauses had only one subject / pattern.